toolbox/fdc3-for-web/fdc3-web-impl/src/handlers/IntentHandler.ts

```diff
@@ -502,20 +545,33 @@ export class IntentHandler implements MessageHandler {
       );
     }
 
-    const target = arg0.payload.app!!;
-    if (target?.instanceId) {
-      return this.raiseIntentRequestToSpecificInstance(possibleIntentRequests, sc, target);
-    } else if (target?.appId) {
-      return this.raiseIntentRequestToSpecificAppId(possibleIntentRequests, sc, target);
+    const target = arg0.payload.app;
+    if (target) {
+      if (isFullAppIdentifier(target)) {
+        return this.raiseIntentRequestToSpecificInstance(possibleIntentRequests, sc, target);
+      } else if (target.appId) {
+        return this.raiseIntentRequestToSpecificAppId(possibleIntentRequests, sc, target);
+      } else {
+        //invalid target
+        console.warn('Received an invalid target argument for raiseIntentForContext', target, arg0);
+        return errorResponseId(
+          sc,
+          arg0.meta.requestUuid,
+          from,
+          ResolveError.TargetAppUnavailable,
+          'raiseIntentForContextResponse'
+        );
+      }
     } else {
+      //No target
       return this.raiseIntentToAnyApp(possibleIntentRequests, sc);
     }
   }
 
   async findIntentsByContextRequest(
     r: FindIntentsByContextRequest,
     sc: ServerContext<AppRegistration>,
-    from: AppIdentifier
+    from: FullAppIdentifier
   ): Promise<void> {
     // TODO: Add result type
     const { context } = r.payload;
```

Comment:
some coverage missing in this function - you might want to investigate

Reply:
ug, theres more than an uncovered line to sort here - the tests don't cover multiple results for findIntentsByContext and are not picking up the fact that every app with an intent registered is being returned here (filtering is not working). Also running apps are not returned.

Will have a go at fixing it

Reply:
running apps were returned in the past, fwiw

Reply:
only in findIntent, looks like it was missing in findIntentForContext.

Incidentally on findIntent it looks like, if there are instances running, it only returns those and not the the record with just the appId (indicating launch a new instance). I'll confirm that and adjust the tests if thats the case - if it is the case we must be missing that in the conformance tests...

Reply:
fixed both findIntent and findIntentForContext and added tests